### ISSUE TITLE
Next.js with typescript config

### ISSUE THREAD

**Issue by kopeclu2** [orestbida/cookieconsent#303]
Hi there,
can someone give me advice to connect this library with Next.js ? Always when configuring this library, "window is not defined" is thrown. Cannot find solutions and and i spent a lot of hours for build this.
Could you give me advice to built it with Next ? 😔
Thank you so much 😊

**Comment by kopeclu2** [comment]
<img width="1190" alt="Snímek obrazovky 2022-05-08 v 17 28 12" src="https://user-images.githubusercontent.com/43535871/167303329-7ae75243-32b2-4d43-9d8b-29a6fcdfd956.png">

**Comment by orestbida** [comment]
Hi @kopeclu2, 

did you check the "How to use in React" [FAQ](https://github.com/orestbida/cookieconsent#faq) section? 

You need to call the plugin inside the `useEffect` hook to make sure that the window object exists.

**Comment by kopeclu2** [comment]
yes but in FE client! When Next building static site window does not exists ! And look this library does not have null check for window 😊 When i edited library now its working. So not my fault by library is not null safe 😔 
<img width="723" alt="Snímek obrazovky 2022-05-08 v 18 05 44" src="https://user-images.githubusercontent.com/43535871/167304777-82e4ab06-de5d-4ffc-9237-c2db13be93d2.png">

**Comment by kopeclu2** [comment]
And sry i checked FAQ with React but Next is different approach 😔

**Comment by orestbida** [comment]
Next.js does have the `useEffect` hook, and I'll keep recommending running the plugin inside it, as it is — imo — the safest approach. I've tried the same setup in React (pure react), nextjs and gatsby and all worked the same way. Nevertheless, glad you found a solution.

**Comment by kopeclu2** [comment]
Look i use all this approaches but error is in cookieconsent.js. When Next building sites, then uses this script where is window object, which does not exists when Next is building on server. So where is the problem ? I use it right way :/ 

<img width="502" alt="Snímek obrazovky 2022-05-08 v 18 24 12" src="https://user-images.githubusercontent.com/43535871/167305489-57ceae60-cee2-48de-8c52-299c1d301a65.png">

**Comment by orestbida** [comment]
You are right. Took a few minutes to set up a next project with cookieconsent v2.8 and it throws the above error. 

This issue has already been addressed in the v3.0 beta version (not yet production ready).

**Comment by kopeclu2** [comment]
So what should I do ? Use CDN or ? Will error disappear ? Or use Beta ? 😊 Thanks for testing

**Comment by Realme121** [comment]
Hi I’m sorry for all of that misguided information but that wasn’t from me that was from someone else who has been hacking me remotely in my devices I do not consent to the extra cookie policies

Sent from my iPhone

On May 8, 2022, at 11:25 AM, kopeclu2 ***@***.***> wrote:

﻿
Look i use all this approaches but error is in cookieconsent.js. When Next building sites, then uses this script where is window object, which does not exists when Next is building on server. So where is the problem ? I use it right way :/

**Comment by orestbida** [comment]
Unfortunately, v3 beta has some small issues that need to be fixed before it's released (it's not yet available on NPM). I'd recommend staying with the (modified) v2.8 version.

**Comment by ad-si** [comment]
Can you please reopen this issue and only close it once 3.0 is released? I have the same problem and I almost mist this issue because it's closed :/

**Comment by Danilo-A** [comment]
> Can you please reopen this issue and only close it once 3.0 is released? I have the same problem and I almost missed this issue because it's closed :/

Same here

**Comment by orestbida** [comment]
Sure!

**Comment by orestbida** [comment]
This is now fixed in `v2.8.1` and  `v3.0.0-rc.1`.

**Comment by xstable** [comment]
@kopeclu2 

> yes but in FE client! When Next building static site window does not exists ! And look this library does not have null check for window 😊 When i edited library now its working. So not my fault by library is not null safe 😔 <img alt="Snímek obrazovky 2022-05-08 v 18 05 44" width="723" src="https://user-images.githubusercontent.com/43535871/167304777-82e4ab06-de5d-4ffc-9237-c2db13be93d2.png">

Can you please give me a helping Hand?
I try to get CookieConsent to work with NextJS18, and tried this: 
```JS
import { useEffect } from 'react';
import pluginConfig from './CookieConsentConfig';
import 'vanilla-cookieconsent';

const CookieConsentComponent = () => {
    useEffect(() => {
        /**
         * useEffect is executed twice (React 18+),
         * make sure the plugin is initialized and executed once
         */
        if (typeof window !== 'undefined') {
            if (!document.getElementById('cc--main')) {
                window.CC = window.initCookieConsent();
                window.CC.run(pluginConfig);
            }
        }
    }, []);

    return null;
};

export default CookieConsentComponent;
```

But I only got following error: 

![image](https://github.com/orestbida/cookieconsent/assets/1778068/3fd4957e-de29-4b96-b672-4d31b0c41edb)

**Comment by Reagankm** [comment]
@xstable Do you have `import vanilla-cookieconsent' at the top?  

I was getting that same error but did some searches on Github to find people who had successful implementations and that import fixed it for me.

**Comment by xstable** [comment]
Yes, I have the import-file at the top of my component... that's why I am confused, that the script isn't loaded.
I've implemented one by one like described in the "React"-HowTo which is linked from the Readme.md.

![image](https://github.com/orestbida/cookieconsent/assets/1778068/23dcea68-5616-46c9-b2bb-8395df5a90d3)

**Comment by orestbida** [comment]
@xstable please open a new issue/discussion!

**Comment by FreiBj** [comment]
Did you find a solution? I'm getting a very similar error.
<img width="562" alt="image" src="https://github.com/orestbida/cookieconsent/assets/39388734/6f68525c-e5b3-4394-b909-18227e95813d">

Text version:
`./src/app/components/gdpr.tsx:10:11
Type error: Property 'CC' does not exist on type 'Window & typeof globalThis'.

   8 |  useEffect(() => {
   9 |          if (!document.getElementById('cc--main')) {
> 10 |                  window.CC = window.initCookieConsent()
     |                         ^
  11 |                  window.CC.run(pluginConfig)
  12 |          }
  13 |  }, [])
error: script "build" exited with code 1 (SIGHUP)`

**Comment by xstable** [comment]
I wend back to my own implementation. Which is way more complicated, but working at least.

**Comment by orestbida** [comment]
@FreiBj please open a new discussion/issue and provide a small demo with issue so that I can better understand what's going on!
